src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -645,14 +595,14 @@ private string GenerateForObject(TypeGenerationSpec typeMetadata)
                 string? ctorParamMetadataInitFuncSource = null;
                 string? serializeFuncSource = null;
 
-                string propInitMethodName = "null";
+                string propInitMethod = "null";
                 string ctorParamMetadataInitMethodName = "null";
                 string serializeMethodName = "null";
 
                 if (typeMetadata.GenerateMetadata)
                 {
                     propMetadataInitFuncSource = GeneratePropMetadataInitFunc(typeMetadata);
-                    propInitMethodName = $"{typeFriendlyName}{PropInitMethodNameSuffix}";
+                    propInitMethod = $"_ => {typeFriendlyName}{PropInitMethodNameSuffix}({OptionsLocalVariableName})";
```

Comment:
I'm still a bit skeptical about this change. It suggests to me that we're violating the way these metadata APIs were meant to be initialized (i.e. these should be populated using a context argument, but instead we're ignoring that parameter and capturing inputs from the environment). It might not be that important but I don't understand this part of the codebase enough to know if it has the potential of breaking things in a meaningful way.

Reply:
we don't ignore them. If it's assigned it will be used. We do not assign it though because combinator would use JsonTypeInfo user would not expect. We previously could do this shortcut because it wasn't possible to get different answer. Now that it's possible user would expect to be able to tweak it.